File: programming/Programming Challenges/2. Minesweeper.c
#include<stdio.h>
#include<conio.h>

int max(int a,int b );
int min(int a,int b );

void main()
{
	int n,m,i,j,a,b,mines,count=0;
	scanf("%d %d",&n,&m);
	while(n!=0||m!=0)
	{
		count+=1;
		char field[n][m];
		for (i=0;i<n;i++)
		{
			scanf("%s",&field[i]);
		}
		char uncovered[n][m+1];
		for (i=0;i<n;i++)
		{
			for (j=0;j<m;j++)
			{
				mines=0;
				if (field[i][j]=='.')
				{
					for (a=max(0,i-1);a<min(i+2,n);a++)
					{
						for(b=max(0,j-1);b<min(j+2,m);b++)
						{
							if(field[a][b]=='*')
							{
								mines+=1;
							}
						}
					}
					uncovered[i][j]='0'+mines;
				}
				else
				{
					uncovered[i][j]='*';
				}
				uncovered[i][j+1]='\n';
			}
		}
		uncovered[n-1][m]='\0';
		printf("Field #%d:\n",count);
		printf("%s\n\n",uncovered);
		scanf("%d %d",&n,&m);
	}
}

max(a,b)
{return a>b?a:b;}
min(a,b)
{return a<b?a:b;}

File: programming/Programming Challenges/1. Getting Started/2. Minesweeper.py
count=0
z=[]
x=raw_input("")
n=int(x[:x.find(" ")])
m=int(x[x.find(" ")+1:])

while(n!=0 or m!=0):
    field=[]
    field_uncovered=""
    count+=1
    z.append("")
    
    for i in list(range(n)):
        field.append(raw_input())
        
    for i in list(range(n)):
        for j in list(range(m)):
            mine_count=0
            if field[i][j]=='.':
                for a in list(range(max(0,i-1),min(n,i+2))):
                    for b in list(range(max(0,j-1),min(m,j+2))):
                        if field[a][b]=='*':
                            mine_count+=1
                field_uncovered+=str(mine_count)
            else:
                field_uncovered+='*'
        field_uncovered+='\n'

    z[count-1]+='Field #'+str(count)+':\n'+field_uncovered

    x=raw_input("")
    n=int(x[:x.find(" ")])
    m=int(x[x.find(" ")+1:])

for i in list(range(max(0,count-1))):
    print z[i]

if count>0:
    print z[count-1][:-1]

raw_input('Press any key to exit...')
